src/transformers/generation/utils.py

```diff
@@ -1154,7 +1175,7 @@ def _validate_assistant(self, assistant_model):
                     "Ensure you load the assistant with the correct encoder-decoder class, e.g. `AutoModelForSpeechSeq2Seq` for Whisper."
                 )
 
-        if not self.config.vocab_size == assistant_model.config.vocab_size:
+        if assistant_tokenizer is None and self.config.vocab_size != assistant_model.config.vocab_size:
             raise ValueError("Make sure the main and assistant model use the same tokenizer")
```

Comment:
Can we modify this message to suggest using your technique? (i.e. to pass the assistant tokenizer as well 🤗 )